src/QueryBuilder.ts

```diff
@@ -133,4 +133,20 @@ export default class QueryBuilder<T extends IEntity>
   find(): Promise<T[]> {
     return this.executor.execute(this.queries, this.limitVal, this.orderByObj);
   }
+
+  findOne(): Promise<T | null> {
+    const execution = this.executor.execute(this.queries, this.limitVal, this.orderByObj, true);
+    
+    return new Promise<T | null>((resolve, reject) => {
+      execution.then(singleOrNone => {
+        if (singleOrNone.length === 1) {
+          resolve(singleOrNone[0]);
+        } else {
+          resolve(null);
+        }
+      }).catch(e => {
+        reject(e);
+      });
+    });
+  }
 }
```

Comment:
Great thinking by calling the `execute` method! I think this method can be written in a better way by using the power of `async/await`. Something in the lines of 

```typescript
  async findOne(): Promise<T | null> {
    const queryResult = await this.executor.execute(this.queries, this.limitVal, this.orderByObj, true);

    return queryResult.length ? queryResult[0] : null;
}
```